src/QMCDrivers/QMCDriverNew.cpp

```diff
@@ -485,37 +468,60 @@ std::ostream& operator<<(std::ostream& o_stream, const QMCDriverNew& qmcd)
 
 void QMCDriverNew::defaultSetNonLocalMoveHandler(QMCHamiltonian& ham) {}
 
-QMCDriverNew::AdjustedWalkerCounts QMCDriverNew::adjustGlobalWalkerCount(Communicate* comm,
-                                                                         IndexType desired_count,
+QMCDriverNew::AdjustedWalkerCounts QMCDriverNew::adjustGlobalWalkerCount(int num_ranks,
+                                                                         int rank_id,
+                                                                         IndexType required_total,
                                                                          IndexType walkers_per_rank,
                                                                          RealType reserve_walkers,
                                                                          int num_crowds)
 {
-  int ranks = myComm->size();
-  AdjustedWalkerCounts awc{0, 0, 0, 0};
-  awc.global_walkers   = desired_count;
-  awc.walkers_per_rank = walkers_per_rank;
-  awc.reserve_walkers  = reserve_walkers;
-  if (awc.global_walkers != 0)
+  AdjustedWalkerCounts awc{required_total, {}, {}, reserve_walkers};
+
+  if (required_total != 0)
   {
-    if (awc.global_walkers % ranks)
+    awc.walkers_per_rank = fairDivide(required_total, num_ranks);
+    if (walkers_per_rank != 0)
     {
-      awc.walkers_per_rank = awc.global_walkers / ranks + 1;
-      awc.global_walkers   = awc.walkers_per_rank * ranks;
-      app_warning() << "TargetWalkers not divisible by number of ranks, TargetWalker increased to "
-                    << awc.global_walkers << '\n';
+      if (std::any_of(awc.walkers_per_rank.begin(), awc.walkers_per_rank.end(),
+                      [walkers_per_rank](IndexType num_walkers) { return num_walkers != walkers_per_rank; }))
```

Comment:
We can simply check "required_total / walkers_per_rank * walkers_per_rank == required_total"?

Reply:
Makes sense to me. Done